### DISCUSSION TITLE
[VOTE] Roadmap - Remove laravel collective dependency

### DISCUSSION THREAD

**Discussion by misterdebug** [misterdebug/crud-generator-laravel#33] [Polls]
All forms are handled by laravel collective package.

**Comment by misterdebug** [comment]
If you have a preferred approach for it, input components, etc., we can discuss it here. It's going to be a significant project for the arrival of L11.
Code examples are welcome :)

**Comment by duganwdobbs** [comment]
I am primarily a Ruby/Rails guy, so a lot of my thoughts are tainted that way, and the package I use there is https://github.com/codez/dry_crud. They have a form builder class that doesn't use templates, but this makes things a lot more complicated on the code side. 

I don't think its a reasonable expectation to go that route, and I'm not sure about the LOE required to build out the templates for all the input fields.

Going with who would be using the package, I believe it would primarily be devs looking to do a quick bootstrap of a project, and a lot of Spatie's libraries lend to that, menus, html generation, etc. I don't think its unreasonable to have Spatie's html lib as a dependency.

Coming from the Rails side, PHP devs seem to want to code their own things rather than use someone else's, so I really don't see any value on building this out a lot more, will be difficult to find more people to help.

**Comment by misterdebug** [comment]
Natively, Laravel does not include a form builder, unlike Symfony, which I personally prefer. The issue with having a dependency is that it may become obsolete over time (see Laravel Collective), especially when it's not truly necessary. Generating a form is far from complicated, so for me, relying on a package for that is a drawback.

I'm trying to make this CRUD Generator independent of any dependencies or libraries (JavaScript, etc.), and removing this last dependency would be much better in my opinion :)

But, of course, this is just my opinion, and I'm open to other perspectives. We still have some time before the arrival of L11 💪

**Comment by duganwdobbs** [comment]
I mean the way I see it you have two options, during your initial install / stub publish you publish your partials for the inputs, or you write the code into the generated templates when you do the crud generation. I think the second option would be more in line with the name of the library.

**Reply by misterdebug** [reply]
I don't think about partials or templates but blade components
Something like `<x-form-input name="name" label="Name" />` (just an example) and will be generated by the crud

**Comment by andrewanswer** [comment]
Now the L11 is out, so I think that best way is to change dependency to LaravelLux/html package which have L11 support, and release the new version.
